homeassistant/components/shelly/config_flow.py

```diff
@@ -37,19 +38,20 @@ async def validate_input(hass: core.HomeAssistant, host, data):
 
     Data has the keys from DATA_SCHEMA with values provided by the user.
     """
+    ip_address = await hass.async_add_executor_job(gethostbyname, host)
+
     options = aioshelly.ConnectionOptions(
-        host, data.get(CONF_USERNAME), data.get(CONF_PASSWORD)
+        ip_address, data.get(CONF_USERNAME), data.get(CONF_PASSWORD)
     )
-    coap_context = await aiocoap.Context.create_client_context()
+    coap_context = await get_coap_context(hass)
+
     async with async_timeout.timeout(5):
         device = await aioshelly.Device.create(
             aiohttp_client.async_get_clientsession(hass),
             coap_context,
             options,
         )
 
-    await coap_context.shutdown()
```

Comment:
You still need to shut down the device here. Or else it will remain attached to the coap context. `device.shutdown()`